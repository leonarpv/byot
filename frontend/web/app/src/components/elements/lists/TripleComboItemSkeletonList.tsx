import * as React from 'react';
import {Grid} from '@material-ui/core';
import * as _ from 'lodash';
import {TripleComboItemSkeleton} from '../list-items/TripleComboItemSkeleton';

interface Props {
  skeletons?: number;
}

export const TripleComboItemSkeletonList: React.FC<Props> = (props: Props) => {
  return (
    <Grid container spacing={1}>
      {_.times(props.skeletons || 10, () => (
        <Grid item xs={12}>
          <TripleComboItemSkeleton />
        </Grid>
      ))}
    </Grid>
  );
};
