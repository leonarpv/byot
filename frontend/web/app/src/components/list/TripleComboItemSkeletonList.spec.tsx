import * as React from 'react';
import {render} from '@testing-library/react';
import {TripleComboItemSkeletonList} from './TripleComboItemSkeletonList';

jest.mock('../elements/list-items/TripleComboItemSkeleton', () => ({
  __esModule: true,
  TripleComboItemSkeleton: () => <div data-testid="skeleton" />,
}));

describe('<TripleComboItemSkeletonList/>', () => {
  it('should render with corresponding count', () => {
    const {container, queryAllByTestId} = render(<TripleComboItemSkeletonList skeletons={12} />);
    expect(queryAllByTestId('skeleton')).toHaveLength(12);
    expect(container).toMatchSnapshot();
  });
});
