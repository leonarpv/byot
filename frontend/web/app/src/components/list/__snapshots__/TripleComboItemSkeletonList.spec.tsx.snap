// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripleComboItemSkeletonList/> should render with corresponding count 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        data-testid="skeleton"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        data-testid="skeleton"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        data-testid="skeleton"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        data-testid="skeleton"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        data-testid="skeleton"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        data-testid="skeleton"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        data-testid="skeleton"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        data-testid="skeleton"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        data-testid="skeleton"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        data-testid="skeleton"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        data-testid="skeleton"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        data-testid="skeleton"
      />
    </div>
  </div>
</div>
`;
